### PR TITLE
Make localstorage draft respect newer saved draft

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)
- [x] Feature
- [x] Bug Fix

## Description
[what the bug currently looks like](https://youtu.be/vlbflkoaRgo)

Our frontend app will now check if the locally cached draft's timestamp is newer or older than the draft provided by the backend before attempting to overwrite it. This is fixed by also caching the timestamp of when localstorage draft is created, then the timestamp is compared to decide if the override should occur.

## Related Tickets & Documents
Resolves https://github.com/thepracticaldev/dev.to/issues/3670
## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
TBD
## Added tests?
- [x] yes

## Added to documentation?
- [x] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?
n/a

### CODE REVIEW

File: app/javascript/article-form/articleForm.jsx [hunk 1]
```diff
@@ -139,25 +143,17 @@ export default class ArticleForm extends Component {
     }
   }
 
-  checkContentChanges = previousContent => {
```
Comment (maestromac) [inline app/javascript/article-form/articleForm.jsx]: I can't imagine the case where this is still needed. I believe `isLocalstorageNewer` would handle everything.

**Comment by benhalpern** [comment]
Yay! Really excited to deal with this case.

**Comment by maestromac** [comment]
@JoshCheek are you using the v1 editor? This is from my v2 (`rich + markdown`)

image

**Comment by joshpuetz** [comment]
Ah, I was indeed on the v1 editor, thanks @maestromac !

I couldn't reproduce the error flow outlined in https://github.com/thepracticaldev/dev.to/issues/3670 however: whenever I click on `Save Draft`, I'm taken back to my dashboard and my draft is saved. 

Can anybody else actually reproduce the bug?

**Comment by maestromac** [comment]
@joshpuetz I've updated the PR with what the bug current looks like.

**Comment by joshpuetz** [comment]
Ah, thanks @maestromac, that helps explain it! (Also, it's wild that the editor saves drafts locally as you type: there's no UI indication that it's doing so!)

File: app/javascript/article-form/articleForm.jsx (hunk 1 above)
Comment (rhymes) [reply app/javascript/article-form/articleForm.jsx]: the only scenario i can see this is collaborative realtime editing but we aren't there yet 🤣

**Pull request merged**